proof/crefine/AARCH64/Schedule_C.thy

```diff
@@ -75,14 +75,33 @@ crunch vcpuSwitch
   for pspace_canonical'[wp]: pspace_canonical'
   (wp: crunch_wps)
 
+\<comment> \<open>The C and specifications differ in where the lazyFpuRestore is located within switchToThread; in
+   the specifications it is in ARCH.switchToThread while in the C it is directly in the generic
+   switchToThread. This difference is because the C treats FPUs being enabled as a generic feature,
+   while it is an architecture feature in the specifications.
+   To make things line up in switchToThread_ccorres we rewrite ARCH.switchToThread to have the
+   lazyFpuRestore outside of it.\<close>
+definition arch_switchToThread where
+  "arch_switchToThread tcb \<equiv> (do
+     tcbobj \<leftarrow> getObject tcb;
+     vcpuSwitch (atcbVCPUPtr $ tcbArch tcbobj);
+     setVMRoot tcb
+   od)"
+
+lemma arch_switchToThread_rewrite:
```

Comment:
does the _rewrite name fit into any kind of pattern of how we did these before? rewrite is kinda generic

Reply:
In general `rewrite` is a bit too vague, but since `arch_switchToThread` is just defined above for begin rewritten into `switchToThread` I would be fine with this one.

Reply:
I couldn't find anything else doing quite the same thing to copy a naming pattern from and just went with rewrite, hoping that it would make sense with the comment I included. It's not a lemma that's intended to be used or found outside of this one specific use case.